src/rabbitmq.c

```diff
@@ -27,19 +27,124 @@
 #define  _GNU_SOURCE
 #endif
 
+#include <stdio.h>
 #include <stdbool.h>
+#include <sys/time.h>
+#include <errno.h>
+#include <ell/ell.h>
+#include <hal/linux_log.h>
+#include <amqp.h>
+#include <amqp_tcp_socket.h>
 
 #include "settings.h"
 #include "rabbitmq.h"
 
+static amqp_connection_state_t conn;
+
+static const char *amqp_server_exception_string(amqp_rpc_reply_t reply)
+{
+	amqp_connection_close_t *m = reply.reply.decoded;
+	static char r[512];
+	switch (reply.reply.id) {
+	case AMQP_CONNECTION_CLOSE_METHOD:
+		snprintf(r, sizeof(r),
+			 "server connection error %uh, message: %.*s\n",
+			 m->reply_code, (int)m->reply_text.len,
+			 (char *)m->reply_text.bytes);
+		break;
+	case AMQP_CHANNEL_CLOSE_METHOD:
+		snprintf(r, sizeof(r),
+			 "server channel error %uh, message: %.*s\n",
+			 m->reply_code, (int)m->reply_text.len,
+			 (char *)m->reply_text.bytes);
+		break;
+	default:
+		snprintf(r, sizeof(r),
+			 "unknown server error, method id 0x%08X\n",
+			 reply.reply.id);
+		break;
+	}
+
+	return l_strdup(r);
+}
+
+static const char *amqp_rpc_reply_string(amqp_rpc_reply_t reply)
+{
+	switch (reply.reply_type) {
+	case AMQP_RESPONSE_NONE:
+		return "missing RPC reply type!";
+	case AMQP_RESPONSE_LIBRARY_EXCEPTION:
+		return amqp_error_string2(reply.library_error);
+	case AMQP_RESPONSE_SERVER_EXCEPTION:
+		return amqp_server_exception_string(reply);
+	case AMQP_RESPONSE_NORMAL:
+	default:
+		return "";
+	}
+	return "";
+}
+
 int rabbitmq_start(struct settings *settings)
 {
-	// TODO: connect with rabbitmq-server
+	amqp_socket_t *socket = NULL;
```

Comment:
"I think there is no right or wrong at this point but initialize a pointer to NULL can be a safeguard against possible bugs since a lot of code already verifies if pointers are NULL."

Based on [here](https://cboard.cprogramming.com/c-programming/129311-initializing-pointers-always-necessary.html#post965362)

But... I think this practice isn't common in this service ([ex1](https://github.com/CESARBR/knot-service-source/blob/master/src/device.c#L520), [ex2](https://github.com/CESARBR/knot-service-source/blob/master/src/device.c#L520), [ex3](https://github.com/CESARBR/knot-service-source/blob/master/src/device.c#L520)).